### PR TITLE
DM-17376: Remove global cache

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: policy/cameraTransforms.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
```
Comment (timj) [inline policy/cameraTransforms.yaml]: Please use the updated preamble (look in the existing files).

File: python/lsst/obs/lsst/auxTel.py [hunk 2]
```diff
@@ -119,8 +86,7 @@ class AuxTelParseTask(LsstCamParseTask):
     translation methods.
     """
 
-    _cameraClass = AuxTelCam           # the class to instantiate for the class-scope camera
-    _translatorClass = LsstAuxTelTranslator
```
Comment (timj) [inline python/lsst/obs/lsst/auxTel.py]: This `_translatorClass` is really important and has to be restored.

File: python/lsst/obs/lsst/imsim.py [hunk 3]
```diff
@@ -57,25 +37,11 @@ class ImsimMapper(LsstCamMapper):
...
-        return ImsimCam()
+    _cameraName = "imsim"
 
 
 class ImsimParseTask(LsstCamParseTask):
     """Parser suitable for phosim data.
     """
 
-    _cameraClass = ImsimCam           # the class to instantiate for the class-scope camera
-    _translatorClass = ImsimTranslator
```
Comment (timj) [inline python/lsst/obs/lsst/imsim.py]: Restore this `_translatorClass` line.

File: python/lsst/obs/lsst/lsstCamMapper.py [hunk 4]
```diff
@@ -147,30 +147,19 @@ def assemble_raw(dataId, componentInfo, cls):
...
-        rotangle = visitInfo.getBoresightRotAngle()
-
-        if boresight.isFinite():
-            exposure.setWcs(getWcsFromDetector(_camera, exposure.getDetector(), boresight,
-                                               90*afwGeom.degrees - rotangle))
-        else:
-            # Should only warn for science observations but VisitInfo does not know
-            logger.warn("Unable to set WCS for %s from header as RA/Dec/Angle are unavailable" %
-                        (dataId,))
+    visitInfo = LsstCamMakeRawVisitInfo(logger)(md, exposureId=-1)
```
Comment (timj) [inline python/lsst/obs/lsst/lsstCamMapper.py]: This isn't guaranteed to work because it all depends on which translator class `LsstCamMakeRawVisitInfo` is using. At the moment the translator class is set to None so the system will work it out from the headers. This is meant to be fine but assumes that you don't decide to force a true LsstCam translator into that class when we get real data arriving.  You are ideally meant to use the ImsimMakeRawVisitInfo and AuxTelMakeRawVisitInfo classes etc.

This code did work before by reading the VisitInfo from the exposure (I checked the values being read) so why is the code now recalculating the VisitInfo?

File: python/lsst/obs/lsst/lsstCamMapper.py [hunk 5]
```diff
@@ -302,6 +260,20 @@ def __init__(self, inputPolicy=None, **kwargs):
             raise RuntimeError("You have more filters defined than fit into the %d bits allocated" %
                                LsstCamMapper._nbit_filter)
 
+    @classmethod
+    def getCameraName(cls):
+        return cls._cameraName
+
+    @classmethod
+    def _makeCamera(cls, policy=None, repositoryDir=None, cameraYamlFile=None):
+        """Make a camera (instance of lsst.afw.cameraGeom.Camera) describing the camera geometry."""
```
Comment (timj) [inline python/lsst/obs/lsst/lsstCamMapper.py]: Can you update the docstring here to match the numpydoc format (you can copy and paste from the versions that you deleted in the subclasses).

File: python/lsst/obs/lsst/lsstCamMapper.py [hunk 6]
```diff
@@ -320,16 +292,6 @@ def defineFilters(cls):
         # official y filter
         afwImageUtils.defineFilter('y', lambdaEff=971.68, lambdaMin=975.0, lambdaMax=1075.0, alias=['y4'])
 
-    def _makeCamera(self, policy, repositoryDir):
-        """Make a camera  describing the camera geometry.
```
Comment (timj) [inline python/lsst/obs/lsst/lsstCamMapper.py]: For example this docstring can move up to the replacement _makeCamera above.

File: python/lsst/obs/lsst/lsstCamMapper.py [hunk 7]
```diff
@@ -546,4 +508,4 @@ def std_raw(self, item, dataId, filter=True):
         """Standardize a raw dataset by converting it to an
         `~lsst.afw.image.Exposure` instead of an `~lsst.afw.image.Image`."""
         return self._standardizeExposure(self.exposures['raw'], item, dataId,
-                                         trimmed=False, setVisitInfo=True, filter=filter)
+                                         trimmed=False, setVisitInfo=False, filter=filter)
```
Comment (timj) [inline python/lsst/obs/lsst/lsstCamMapper.py]: Is this why the VisitInfo is not set? The commit history doesn't really tell me why this change was made.

File: python/lsst/obs/lsst/phosim.py [hunk 8]
```diff
@@ -57,24 +37,41 @@ class PhosimMapper(LsstCamMapper):
...
+    _cameraName = "phosim"
+
+
+class PhosimParseTask(LsstCamParseTask):
+    """Parser suitable for phosim data.
+    """
+
+    _mapperClass = PhosimMapper
+
+    def translate_filter(self, md):
```
Comment (timj) [inline python/lsst/obs/lsst/phosim.py]: Remove this method.

File: python/lsst/obs/lsst/phosim.py [hunk 9]
```diff
@@ -57,24 +37,41 @@ class PhosimMapper(LsstCamMapper):
...
+        filter : `str`
+            filter name
         """
-        return PhosimCam()
+        return md.get("FILTER")
 
-    @classmethod
-    def getCameraName(cls):
-        return 'phosim'
+    def translate_visit(self, md):
```
Comment (timj) [inline python/lsst/obs/lsst/phosim.py]: Remove this method.

File: python/lsst/obs/lsst/phosim.py [hunk 10]
```diff
@@ -57,24 +37,41 @@ class PhosimMapper(LsstCamMapper):
...
+        """Extract visit from metadata (as an int)
 
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList or PropertySet`
+            image metadata
 
-class PhosimParseTask(LsstCamParseTask):
```
Comment (timj) [inline python/lsst/obs/lsst/phosim.py]: Restore this class. The rebase messed up in this file.

File: python/lsst/obs/lsst/ts8.py [hunk 11]
```diff
@@ -22,27 +22,16 @@
...
-    Parameters
-    ----------
-    cameraYamlFile : `str`, optional
-        Path to camera YAML file. Will default to one in this package.
-    """
-    packageName = 'obs_lsst'
-
-    def __init__(self, cameraYamlFile=None):
-        pass
+def computeVisit(dateObs):
```
Comment (timj) [inline python/lsst/obs/lsst/ts8.py]: Remove computeVisit.

File: python/lsst/obs/lsst/ts8.py [hunk 12]
```diff
@@ -52,24 +41,11 @@ class Ts8MakeRawVisitInfo(LsstCamMakeRawVisitInfo):
 
 class Ts8Mapper(LsstCamMapper):
     """The Mapper for the ts8 camera."""
-    MakeRawVisitInfoClass = Ts8MakeRawVisitInfo
```
Comment (timj) [inline python/lsst/obs/lsst/ts8.py]: This class definition needs to be restored.

File: python/lsst/obs/lsst/ts8.py [hunk 13]
```diff
@@ -110,8 +86,7 @@ class Ts8ParseTask(LsstCamParseTask):
     translation methods.
     """
 
-    _cameraClass = Ts8           # the class to instantiate for the class-scope camera
-    _translatorClass = LsstTS8Translator
```
Comment (timj) [inline python/lsst/obs/lsst/ts8.py]: This line needs to come back.

File: python/lsst/obs/lsst/utils.py [hunk 14]
```diff
@@ -29,16 +29,9 @@
...
-    from lsst.obs.lsst.lsstCamMapper import _camera
-except ImportError:
-    from lsst.obs.lsst.auxTel import AuxTelMapper
-    mapper = AuxTelMapper()    # Sets the sleazy _camera global
 
-    from lsst.obs.lsst.lsstCamMapper import _camera
 
-
-def readRawFile(fileName, dataId={}):
+def readRawFile(fileName, dataId={}, detector=None):
```
Comment (timj) [inline python/lsst/obs/lsst/utils.py]: Add `detector` to the docstring for this function.

File: bin.src/generateCamera.py [hunk 15]
```diff
@@ -64,6 +64,10 @@ def findYamlOnPath(fileName, searchPath):
     with open(cameraSklFile) as fd:
         cameraSkl = yaml.load(fd, Loader=yaml.Loader)
 
+    cameraTransformsFile = findYamlOnPath("cameraTransforms.yaml", searchPath)
+    with open(cameraTransformsFile) as fd:
+        cameraTransforms = yaml.load(fd, Loader=yaml.Loader)
```
[the remark this answers is not in the document]
Comment (RobertLuptonTheGood) [reply bin.src/generateCamera.py]: Ignorance or history.  Fixed

File: python/lsst/obs/lsst/phosim.py (hunk 10 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/lsst/phosim.py]: I assume that this means restore to the trivial one on master;  done (and added _mapperClass)

Comment (timj) [reply python/lsst/obs/lsst/phosim.py]: Yes. It just needs to define the translator class.

File: python/lsst/obs/lsst/auxTel.py [hunk 16]
```diff
@@ -57,23 +38,9 @@ class AuxTelMakeRawVisitInfo(LsstCamMakeRawVisitInfo):
 
 class AuxTelMapper(LsstCamMapper):
     """The Mapper for the auxTel camera."""
-    MakeRawVisitInfoClass = AuxTelMakeRawVisitInfo
```
Comment (timj) [inline python/lsst/obs/lsst/auxTel.py]: This line needs to come back I think.

File: python/lsst/obs/lsst/lsstCamMapper.py (hunk 7 above)
Comment (RobertLuptonTheGood) [reply python/lsst/obs/lsst/lsstCamMapper.py]: We've already set the `VisitInfo` and the metadata got stripped.  This may be a change of behaviour -- I hope this all goes away when composites don't have to handle setting their own VisitInfo.  I'll add a comment

Comment (timj) [reply python/lsst/obs/lsst/lsstCamMapper.py]: VisitInfo creation has always stripped the metadata. In the past you may not have noticed because VisitInfo barely had anything in it in obs_lsst.
